src/IROperator.h

```diff
@@ -1907,6 +1909,51 @@ inline Expr likely_if_innermost(Expr e) {
                                 {e}, Internal::Call::PureIntrinsic);
 }
 
+
+/** Cast an expression to the halide type corresponding to the C++
+ * type T clamping to the minimum and maximum values of the result
+ * type. */
+template <typename T>
+Expr saturating_cast(Expr e) {
+    return saturating_cast(type_of<T>(), e);
+}
+
+/** Cast an expression to a new type, clamping to the minimum and
+ * maximum values of the result type. */
+inline Expr saturating_cast(Type t, Expr e) {
+    // For float to float, guarantee infinities are always pinned to range.
```

Comment:
Avoiding +/- Inf is probably good, since (per discussion at https://github.com/halide/Halide/issues/1440 and https://github.com/halide/Halide/pull/1420), our promise of -ffast-math implies -ffinite-math-only, which means that we aren't promising reliable handling of non-finite values in the first place.